design/one-pager-terrajet-metadata-extraction.md

```diff
@@ -0,0 +1,272 @@
+# Metadata Extraction from Terraform Registry for Terrajet-based providers
+* Owner: Alper Rifat Uluçınar (@ulucinar)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+### Background
+For providers generated using [Terrajet], the number of managed resources can
+exceed [several hundreds](provider-jet-aws-preview), and especially for the big
+three Terrajet-based providers ([provider-jet-aws], [provider-jet-gcp] and
+[provider-jet-azure]), it's very inconvenient and time consuming to manually
+author example manifests for all those generated resources. The convention we
+have adopted so far is to manually add example manifests for the resources we
+explicity configure in their respective pull requests. 
+
+Another dimension we need to consider is that currently we are lacking API
+documentation for the generated resources. Although it's possible to redirect
+users of those APIs to the [Terraform registry], it's desirable to have the
+documentation generated together with the API (as comments on the associated
+`struct`s and fields), and have them published on `doc.crds.dev`.
+
+There is also a wealth of metadata that we can use to enrich the Terrajet-based
+providers and the generated resources such as category names for the Terraform
+resources. For example, a provider implementation may opt to use the category
+names to group respective APIs. Or the examples provided in the Terraform
+registry hint at reference fields that can help us in auto-generating
+cross-resource references (that appear in those HCL configurations).
+
+While working on generating example manifests for the big three Terrajet-based
+providers `provider-jet-aws`, `provider-jet-gcp` and `provider-jet-azure` in the
+context of the corresponding [Terrajet issue #48], we have seen utility in
+extracting such metadata from the Terraform registry and use it to generate
+example manifests and documentation. In this document, we would like to propose:
```

Comment:
I think `Background` section could be limited to the problem statement in general and the actual proposal could be kept for its own section.

Reply:
Removed proposal hints from the `Background` section as suggested.